internal/distributed/proxy/httpserver/utils.go

```diff
@@ -1748,6 +1748,8 @@ func RequestHandlerFunc(c *gin.Context) {
 	c.Writer.Header().Set("Access-Control-Allow-Credentials", "true")
 	c.Writer.Header().Set("Access-Control-Allow-Headers", "Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization, accept, origin, Cache-Control, X-Requested-With")
 	c.Writer.Header().Set("Access-Control-Allow-Methods", "GET, HEAD, POST, PUT, DELETE, OPTIONS, PATCH")
+	c.Writer.Header().Set("Strict-Transport-Security", "max-age=31536000; includeSubDomains") // Ensures HTTPS usage
```

Comment:
It is recommended that Strict-Transport-Security, max-age, and includeSubDomains be made configurable. This way, open source users can modify them as needed.